### PR TITLE
DOCS: optional doc building dependencies

### LINKED ISSUE #18355: Optional components required to build docs aren't documented
### Problem

Some optional components are required to get the docs building without warnings. AFAICT, these are not mentioned in the [dev docs](https://matplotlib.org/devel/documenting_mpl.html#getting-started). These include:

1. Inkscape
```
WARNING: Inkscape command 'inkscape' cannot be run. Check the inkscape_converter_bin setting
```
2. Fonts
```
findfont: Font family ['xkcd', 'xkcd Script', 'Humor Sans', 'Comic Neue', 'Comic Sans MS'] not found. Falling back to DejaVu Sans.
```
3. Optipng
```
WARNING: optipng binaries not found, PNG thumbnails and images will not be optimized
```

### Suggested Improvement

Once @jeromefv confirms that I'm not missing anything, I can make a PR to update the docs to contain all this information. They probably don't need to be front and center, since the docs build just fine without these (except maybe Inkscape?) But it'd be nice to have an "optional dependencies" note.

### PR DESCRIPTION
**Pull request by brunobeltran**
## PR Summary

Fixes #18355. tl;dr: You need Inkscape, optipng, and Humor Sans fonts installed locally for the docs to build without warnings. 

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by story645 — APPROVED** [review]

File: doc/devel/documenting_mpl.rst [hunk 1]
```diff
@@ -70,16 +70,24 @@ shown below:
 .. include:: ../../requirements/doc/doc-requirements.txt
    :literal:
 
-To (2) set up LaTeX and Graphviz dependencies you should:
+For (2), the Matplotlib docs require that you:
 
 *  install a minimal working LaTeX distribution
-*  install the LaTeX packages cm-super and dvipng
 *  install `Graphviz <http://www.graphviz.org/download>`_
+*  install the LaTeX packages cm-super and dvipng. If your OS bundles `TexLive`,
```
Comment (QuLogic) [inline doc/devel/documenting_mpl.rst:77]: Not a reference.
```suggestion
*  install the LaTeX packages cm-super and dvipng. If your OS bundles ``TexLive``,
```

File: doc/devel/documenting_mpl.rst [hunk 2]
```diff
@@ -70,16 +70,24 @@ shown below:
...
+   then often the "complete" version of the installer will automatically include
+   these packages (e.g. "texlive-full" or "texlive-all").
+*  (suggested) install `Inkscape <https://inkscape.org>`_.
+*  (suggested) install `optipng <http://optipng.sourceforge.net>`_.
+*  (suggested) install the "Humor Sans" font (aka the "XKCD" font).
```
Comment (QuLogic) [inline doc/devel/documenting_mpl.rst:82]: Comic Neue is a Free alternative, too.

**Comment by brunobeltran** [comment]
Thanks @QuLogic for catching the typos. Added Comic Neue as well. 

Should be good to go.

**Review by timhoffm — APPROVED** [review]
Can be improved further. But will do in a follow up PR instead of going back and forth here.

**Pull request merged**